packages/medusa/src/services/customer-group.ts

```diff
@@ -86,6 +89,38 @@ class CustomerGroupService extends BaseService {
     })
   }
 
+  /**
+   * Update a customer group.
+   *
+   * @param {string} customerGroupId - id of the customer group
+   * @param {CustomerGroupUpdate} update - customer group partial data
+   */
+  async update(
+    customerGroupId: string,
+    update: CustomerGroupUpdate
+  ): Promise<CustomerGroup[]> {
+    return this.atomicPhase_(async (manager) => {
+      const { metadata, ...properties } = update
+
+      const cgRepo: CustomerGroupRepository = manager.getCustomRepository(
+        this.customerGroupRepository_
+      )
+
+      const customerGroup = await this.retrieve(customerGroupId)
+
+      for (const key in properties) {
+        if (typeof properties[key] !== "undefined") {
+          customerGroup[key] = properties[key]
+        }
+      }
+
+      if (typeof metadata !== "undefined") {
+        customerGroup.metadata = metadata
+      }
```

Comment:
please use the `setMetadata_` method. This will ensure that we don't overwrite metadata unintentionally. Consider this example:
- a plugin (Plugin A) needs to add a metadata field to a customer group and calls `customerGroupService.update(id, { metadata: { plugin_field: "1234"} })` 
- the plugin uses this field to identify and sync the customer group with a segment in a 3rd party CRM system
- another plugin (Plugin B) also needs to update the metadata field and does: `customerGroupService.update(id, { metadata: { another_field: "1234"} })`
- with the current implementation Plugin A is no longer able to keep systems in sync because the metadata field was overwritten

`setMetadata_` ensures that this doesn't happen.